File: src/main/java/am/petstore/PetStore/pets/controller/PetController.java
package am.petstore.PetStore.pets.controller;

import am.petstore.PetStore.pets.service.PetService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.Map;

@RestController
@RequestMapping("/pets")
public class PetController {

    private final PetService service;

    @Autowired
    public PetController(PetService service) {
        this.service = service;
    }

    @GetMapping("/findAll")
    public ResponseEntity findAll() {
        return service.findAll();
    }


    @PostMapping(value = "/create", consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
    public ResponseEntity<Map<Object, Object>> create(@RequestParam(value = "image", required = false) MultipartFile photo,
                                                      @RequestParam(value = "title", required = false) String title) {
        return service.create(photo, title);
    }

    @PostMapping(value = "/update/{id}", consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
    public ResponseEntity<Map<Object, Object>> update(@PathVariable("id") Long id, @RequestParam(value = "image", required = false) MultipartFile photo,
                                                      @RequestParam(value = "title", required = false) String title) {
        return service.update(id, photo, title);
    }

    @PutMapping("/delete/{id}")
    public ResponseEntity delete(@PathVariable("id") Long id) {
        return service.delete(id);
    }
}

File: src/main/java/am/petstore/PetStore/pets/controller/PetController.kt
package am.petstore.PetStore.pets.controller

import am.petstore.PetStore.pets.service.PetService
import kotlinx.coroutines.FlowPreview
import kotlinx.coroutines.runBlocking
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile
import org.springframework.web.reactive.function.client.WebClient

@RestController
@RequestMapping("/pets")
class PetController @Autowired constructor(private val petService: PetService) {

    @FlowPreview
    @GetMapping("/findAll")
    fun findAll(): ResponseEntity<MutableMap<Any, Any>> {
        return runBlocking {
            petService.findAll()
        }
    }

    @PostMapping(value = ["/create"], consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun create(@RequestParam(value = "image", required = false) photo: MultipartFile?,
               @RequestParam(value = "title", required = false) title: String?): ResponseEntity<Map<Any, Any>>? {
        return petService.create(photo, title)
    }

    @PostMapping(value = ["/update/{id}"], consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun update(@PathVariable("id") id: Long?, @RequestParam(value = "image", required = false) photo: MultipartFile?,
               @RequestParam(value = "title", required = false) title: String?): ResponseEntity<Map<Any, Any>>? {
        return petService.update(id!!, photo, title)
    }

    @PutMapping("/delete/{id}")
    fun delete(@PathVariable("id") id: Long?): ResponseEntity<*>? {
        return petService.delete(id!!)
    }

}